vajram/extensions/sql/vajram-sql-codegen/src/main/java/com/flipkart/krystal/vajram/sql/codegen/AbstractSqlCodeGenerator.java

```diff
@@ -0,0 +1,346 @@
+package com.flipkart.krystal.vajram.sql.codegen;
+
+import com.flipkart.krystal.codegen.common.models.CodegenPhase;
+import com.flipkart.krystal.codegen.common.spi.CodeGenerator;
+import com.flipkart.krystal.vajram.TraitRoot;
+import com.flipkart.krystal.vajram.codegen.common.models.VajramCodeGenUtility;
+import com.flipkart.krystal.vajram.codegen.common.models.VajramInfo;
+import com.flipkart.krystal.vajram.codegen.common.spi.VajramCodeGenContext;
+import com.flipkart.krystal.vajram.sql.annotations.SqlQuery;
+import com.flipkart.krystal.vajram.sql.annotations.SqlUpdate;
+import com.squareup.javapoet.ClassName;
+import com.squareup.javapoet.TypeName;
+import java.util.ArrayList;
+import java.util.List;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.element.VariableElement;
+import javax.lang.model.type.DeclaredType;
+import javax.lang.model.type.TypeKind;
+import javax.lang.model.type.TypeMirror;
+
+public abstract class AbstractSqlCodeGenerator implements CodeGenerator {
```

Comment:
Make this package private if this is not intended to be extended outside this module